x-pack/plugin/ilm/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java

```diff
@@ -126,6 +140,95 @@ public void testCanManageIndexWithNoPermissions() throws Exception {
         });
     }
 
+    public void testSLMWithPermissions() throws Exception {
+        createIndexAsAdmin("index", Settings.builder().put("index.number_of_replicas", 0).build(), "");
+
+        // Set up two roles and users, one for reading SLM, another for managing SLM
+        Request roleRequest = new Request("PUT", "/_security/role/slm-read");
+        roleRequest.setJsonEntity("{ \"cluster\": [\"read_slm\"] }");
+        assertOK(adminClient().performRequest(roleRequest));
+        roleRequest = new Request("PUT", "/_security/role/slm-manage");
+        roleRequest.setJsonEntity("{ \"cluster\": [\"manage_slm\", \"create_snapshot\"]," +
+            "\"indices\": [{ \"names\": [\".slm-history*\"],\"privileges\": [\"all\"] }] }");
+        assertOK(adminClient().performRequest(roleRequest));
+
+        createUser("slm_admin", "slm-pass", "slm-manage");
+        createUser("slm_user", "slm-user-pass", "slm-read");
+
+        final HighLevelClient hlAdminClient = new HighLevelClient(adminClient());
+
+        // Build two high level clients, each using a different user
+        RestClientBuilder builder = RestClient.builder(adminClient().getNodes().toArray(new Node[0]));
+        String token = basicAuthHeaderValue("slm_admin", new SecureString("slm-pass".toCharArray()));
+        configureClient(builder, Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            .build());
+        builder.setStrictDeprecationMode(true);
+        final RestHighLevelClient adminHLRC = new RestHighLevelClient(builder);
+
+        builder = RestClient.builder(adminClient().getNodes().toArray(new Node[0]));
```

Comment:
I don't love reusing variables in general (`builder` here, `token` in the next line) but it's not a major objection.